### PR TITLE
10983: Add Scorecard alias data element to version data

### PR DESCRIPTION
**Pull request by elder-joseph**
## Description
As a developer, I need to add the Scorecard alias data element to the API pull request in GIDS so that the data can be included in version generation for search. This PR includes the `app`, `db`, and `spec` changes necessary for 10983.

The database specific PR can be found here: https://github.com/department-of-veterans-affairs/gibct-data-service/pull/696

Originating Issue: [10983](https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/10983)

## Testing done
CI passes locally, manual checks of columns in database as well.

## Screenshots
Screen Shot 2020-08-05 at 1 03 11 PM

## Acceptance criteria
- [x] The alias data element is retrieved and stored by the API Fetch request within GIDS.
- [x] The alias data element is included in the version generation.
- [x] The comparison tool search functionality includes the scorecard data element when a search is executed.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: app/models/institution.rb [hunk 1]
```diff
@@ -31,7 +31,7 @@ class Institution < ApplicationRecord
     'institution LIKE :starts_with_term',
     'institution = :search_term',
     'city = :search_term',
-    'ialias LIKE :upper_contains_term'
+    'alias LIKE :upper_contains_term'
```
Comment (zurbergram) [inline app/models/institution.rb:34]: This is not needed

File: app/models/institution.rb [hunk 2]
```diff
@@ -51,7 +51,7 @@ class Institution < ApplicationRecord
     'ope6' => { column: :ope6, converter: Ope6Converter },
     'school_system_name' => { column: :f1sysnam, converter: BaseConverter },
     'school_system_code' => { column: :f1syscod, converter: NumberConverter },
-    'alias' => { column: :ialias, converter: BaseConverter },
+    'alias' => { column: :alias, converter: BaseConverter },
```
Comment (zurbergram) [inline app/models/institution.rb:54]: This is not needed

File: app/models/scorecard.rb [hunk 3]
```diff
@@ -21,7 +21,7 @@ class Scorecard < ApplicationRecord
     retention_all_students_ba retention_all_students_otb
     graduation_rate_all_students salary_all_students
     repayment_rate_all_students avg_stu_loan_debt
-    hbcu menonly womenonly relaffil hcm2 pctfloan
+    hbcu menonly womenonly relaffil hcm2 pctfloan alias
```
Comment (zurbergram) [inline app/models/scorecard.rb:24]: This is not needed

File: app/utilities/scorecard_api/service.rb [hunk 4]
```diff
@@ -28,7 +28,8 @@ class Service
       'latest.repayment.3_yr_repayment_suppressed.overall': :repayment_rate_all_students,
       'latest.student.retention_rate.four_year.full_time': :retention_all_students_ba,
       'latest.student.retention_rate.lt_four_year.full_time': :retention_all_students_otb,
-      'latest.student.size': :undergrad_enrollment
+      'latest.student.size': :undergrad_enrollment,
+      'school.alias': :alias
```
Comment (zurbergram) [inline app/utilities/scorecard_api/service.rb:32]: Styling: move this to be with other school properties

File: db/migrate/20200804170036_add_alias_to_institutions.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class AddAliasToInstitutions < ActiveRecord::Migration[5.2]
```
Comment (zurbergram) [inline db/migrate/20200804170036_add_alias_to_institutions.rb:1]: This is not needed

File: db/migrate/20200804170056_add_alias_to_institutions_archives.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class AddAliasToInstitutionsArchives < ActiveRecord::Migration[5.2]
```
Comment (zurbergram) [inline db/migrate/20200804170056_add_alias_to_institutions_archives.rb:1]: This is not needed

File: spec/controllers/v0/institutions_controller_spec.rb [hunk 7]
```diff
@@ -374,7 +374,7 @@ def check_boolean_facets(facets)
     end
 
     it 'search returns results alias' do
-      create(:institution, :independent_study, ialias: 'UIS', version_id: Version.current_production.id)
+      create(:institution, :independent_study, alias: 'UIS', version_id: Version.current_production.id)
```
Comment (zurbergram) [inline spec/controllers/v0/institutions_controller_spec.rb:377]: This is not needed

**Review by zurbergram — CHANGES_REQUESTED** [review]
at https://github.com/department-of-veterans-affairs/gibct-data-service/blob/master/app/models/institution_builder.rb#L318

update the sql to have `ialias = scorecard.alias` instead
```
UPDATE institutions SET #{columns_for_update(Scorecard)}, ialias = scorecard.alias
      FROM scorecards
      WHERE institutions.cross = scorecards.cross
      AND institutions.version_id = #{version_id}
```

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]
I think we also want to remove ialias from institution. I don't want anyone to get confused by the two values.

**Comment by elder-joseph** [comment]
@shawkey-daniel I had considered adding a migration to remove that, but wasnt sure if we wanted to keep it around for some reason or another as the story said "includes" the alias from scorecard. Looks like going with Matt's suggestions here it will remain, but `ialias` will get the value from scorecard instead.

**Review by zurbergram — APPROVED** [review]

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by delli-gatti-michael — APPROVED** [review]

**Review by f1337 — APPROVED** [review]
Looks good to me! :thumbsup:
Shaun The Sheep giving thumbs up and smiling

**Pull request merged**